kythe/cxx/indexer/textproto/plugin.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef KYTHE_CXX_INDEXER_TEXTPROTO_PLUGIN_H_
+#define KYTHE_CXX_INDEXER_TEXTPROTO_PLUGIN_H_
+
+#include <cstdio>
+#include <string>
+
+#include "absl/status/status.h"
+#include "kythe/cxx/common/file_vname_generator.h"
+#include "kythe/cxx/common/indexing/KytheGraphRecorder.h"
+#include "kythe/cxx/common/status_or.h"
+#include "kythe/cxx/indexer/proto/vname_util.h"
+#include "kythe/proto/analysis.pb.h"
+#include "re2/re2.h"
+
+namespace kythe {
+namespace lang_textproto {
+
+// The plugin's interface to the indexer.
+class PluginApi {
```

Comment:
minor: This class should have a virtual destructor and the copy ctor/assignment operator should either be protected or (public and deleted).